Remotion/Web/Core/Resources/RemoveCacheKeyMiddleware.cs

```diff
@@ -0,0 +1,124 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+#if !NET48
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Remotion.Utilities;
+
+namespace Remotion.Web.Resources;
+
+/// <summary>
+/// ASP.NET Core middleware that removes the cache key generated by <see cref="ResourcePathBuilderBasedStaticResourceCacheKeyProvider"/> from the URL.
+/// The URL is rewritten and the cache key is made available to any further middleware.
+/// </summary>
+/// <seealso cref="CachingApplicationBuilderExtensions.UseRemotionStaticFiles"/>
+/// <seealso cref="CachingApplicationBuilderExtensions.UseRemoveCacheKeyMiddleware"/>
+/// <remarks>
+/// The preferred way of using this middleware is <see cref="CachingApplicationBuilderExtensions"/>.<see cref="CachingApplicationBuilderExtensions.UseRemotionStaticFiles"/>.
+/// </remarks>
+public class RemoveCacheKeyMiddleware
+{
+  private const string c_cacheKeyUrlIdentifier = "/cache_";
+  private const string c_removedCacheKeyItemsKey = "RemotionRemovedCacheKey";
+
+  /// <summary>
+  /// Tries to retrieve a cache key from <paramref name="context"/>, which was previously removed using <see cref="RemoveCacheKeyMiddleware"/>.
+  /// The cache key matches the value returned by <see cref="IStaticResourceCacheKeyProvider"/>.
+  /// </summary>
+  public static bool TryGetRemovedCacheKey (HttpContext context, [NotNullWhen(true)] out string? cacheKey)
```

Comment:
do we ever need the actual value or would this be clearer as a simple boolean operation?

Reply:
We don't need it but I think it is a good to leave options open so I would keep it here.